packages/rest/src/lib/utils/utils.ts

```diff
@@ -11,3 +11,18 @@ export function parseResponse(res: Response): Promise<unknown> {
 
 	return res.buffer();
 }
+
+/**
+ * Check whether a request falls under a sublimit
+ * @param bucketRoute The buckets route identifier
+ * @param body The options provided as JSON data
+ * @returns
+ */
+export function isSublimited(bucketRoute: string, body?: unknown): boolean {
+	// TODO: Update for new sublimits
+	// Currently known sublimits:
+	// Editing channel `name` or `topic`
+	return (
+		typeof body === 'object' && body !== null && bucketRoute === `/channels/:id` && ('name' in body || 'topic' in body)
+	);
```

Comment:
In my opinion, instead of this, you should check the route from the get-go, and make the code a bit cleaner. Here's an example, which could also make it more extensible in the future

```ts
// Editing channel `name` or `topic`
if (bucketRoute === '/channels/:id') {
	// we know what body is in this context
	const casted = body as RestPatchAPIChannelJSONBody;
	return ['name', 'topic'].some((key) => key in casted);
}

return false;
```

Reply:
We probably want to do `typeof casted[key] !== 'undefined'` in that case, since `name: undefined` isn't sent due to JSON's limitations.

Reply:
Either way we'd need that typeof check actually 👀